OutSystems/outsystems-datagrid-reactive
# ROU-3106: Arrange Data not working properly when using complex struct…

Fix an issue when handling more complex structures while obtaining the metadata for the generated JSON from the ArrangeData method.


### What was happening
* During a testing session, we found out that the Data Grid was throwing errors when the resulting JSON comes from a more complex structure like a structure with entities as an attribute
* We found out that the issues are inside the ArranjeData and that in these scenarios we’re not removing the prefix “ss“

### What was done
* Changed the method _cleanRecName_ to handle more complex structures when obtaining the metadata


### Test Steps
1. Create a screen with a grid with a data source from an advanced query that is appended to a list with a structure where we have an entity and another structure inside
2. Check that the information is loaded successfully
3. Click in **Deserialize**
4. Check that the data is correct and that the product name "My Personal Desktop" is displayed on a feedback message


### Screenshots
![ROU3106_Preview](https://user-images.githubusercontent.com/29493222/155301702-91acf6ef-2be9-4095-99e5-f3edfc876385.gif)


### Checklist
